### PR TITLE
Upgrade to Flux version v0.23

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

#600 

on /hold till ship the v0.6 patch release

*Description of changes:*

- Add default namespace to eks-a resources pushed to the Git repo, for both create and upgrade cases
- Delete old flux-syste secret before flux bootstrap, to always use ECDSA as the default SSH key algorithm

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/638#" title="Author self-approved">jiayiwang7</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [jiayiwang7]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by mdsgabriel — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/638#" title="Author self-approved">jiayiwang7</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/638#pullrequestreview-807492982" title="Approved">mdsgabriel</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [jiayiwang7,mdsgabriel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/api/v1alpha1/cluster_types.go
Comment (g-gaston) [inline pkg/api/v1alpha1/cluster_types.go:28]: shouldn't this be `eksa-system`? I think I'm missing something here :)

File: pkg/addonmanager/addonclients/upgrader.go
Comment (g-gaston) [inline pkg/addonmanager/addonclients/upgrader.go:45]: Would flux stop working until the secret is recreated? I wonder if we should pause the flux reconcile before this and enable just after finishing the the flux component upgrade

Comment (g-gaston) [inline pkg/addonmanager/addonclients/upgrader.go:45]: We don't need to figure this out now, but question for the future
Will this work once flux has been upgraded to v0.23? Like imagine the user upgrades to 0.23 and we then release a new bundle with v0.24, will this same process work? Or do we need to check somehow if the secret needs to be deleted first?

**Review by g-gaston — COMMENTED** [review]
Flawless

File: pkg/api/v1alpha1/cluster_types.go
Comment (jiayiwang7) [reply pkg/api/v1alpha1/cluster_types.go:28]: so this is the default namespace for eksa resources (cluster, datacenter, machineconfig, gitopsconfig etc)

File: pkg/addonmanager/addonclients/upgrader.go
Comment (jiayiwang7) [reply pkg/addonmanager/addonclients/upgrader.go:45]: no, the `flux bootstrap` cmd thats executed right after will recreate the secrets.

Comment (jiayiwang7) [reply pkg/addonmanager/addonclients/upgrader.go:45]: i believe so, `flux bootstrap` cmd will always generate a ecdsa-sha2-nistp384 SSH key and will update the deploy key on GitHub.

**Pull request merged**
